mr/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -182,48 +188,101 @@ public static boolean dropTable(Configuration conf, Properties props) {
+      default:
+        return Optional.of(CatalogUtil.buildIcebergCatalog(name,
+                getCatalogProperties(conf, name, catalogType), conf));
     }
+  }
 
-    LOG.info("Catalog is not configured");
-    return Optional.empty();
+  /**
+   * Collect all the catalog specific configuration from the global hive configuration.
+   * @param conf a Hadoop configuration
+   * @param catalogName name of the catalog
+   * @param catalogType type of the catalog
+   * @return complete map of catalog properties
+   */
+  private static Map<String, String> getCatalogProperties(Configuration conf, String catalogName, String catalogType) {
+    String keyPrefix = InputFormatConfig.CATALOG_CONFIG_PREFIX + catalogName;
+    Map<String, String> catalogProperties = Streams.stream(conf.iterator())
+            .filter(e -> e.getKey().startsWith(keyPrefix))
+            .collect(Collectors.toMap(e -> e.getKey().substring(keyPrefix.length() + 1), Map.Entry::getValue));
+    return addCatalogPropertiesIfMissing(conf, catalogType, catalogProperties);
+  }
+
+  /**
+   * This method is used for backward-compatible catalog configuration.
+   * Collect all the catalog specific configuration from the global hive configuration.
+   * Note: this should be removed when the old catalog configuration is depracated.
+   * @param conf global hive configuration
+   * @param catalogType type of the catalog
+   * @param catalogProperties pre-populated catalog properties
+   * @return complete map of catalog properties
+   */
+  private static Map<String, String> addCatalogPropertiesIfMissing(Configuration conf, String catalogType,
+                                                                   Map<String, String> catalogProperties) {
+    catalogProperties.putIfAbsent(CatalogUtil.ICEBERG_CATALOG_TYPE, catalogType);
+    if (catalogType.equalsIgnoreCase(HADOOP_CATALOG_TYPE)) {
+      catalogProperties.putIfAbsent(CatalogProperties.WAREHOUSE_LOCATION,
+              conf.get(InputFormatConfig.HADOOP_CATALOG_WAREHOUSE_LOCATION));
+    }
+    return catalogProperties;
+  }
+
+  /**
+   * Return the catalog type based on the catalog name.
+   * <p>
+   * If the catalog name is provided get the catalog type from 'iceberg.catalog.<code>catalogName</code>.type' config.
+   * In case the value of this property is null, return with no catalog definition (Hadoop Table)
+   * </p>
+   * <p>
+   * If catalog name is null, check the global conf for 'iceberg.mr.catalog' property. If the value of the property is:
+   * <ul>
+   *     <li>null/hive -> Hive Catalog</li>
+   *     <li>location -> Hadoop Table</li>
+   *     <li>hadoop -> Hadoop Catalog</li>
+   *     <li>any other value -> Custom Catalog</li>
+   * </ul>
+   * </p>
+   * @param conf global hive configuration
+   * @param catalogName name of the catalog
+   * @return type of the catalog, can be null
+   */
+  private static String getCatalogType(Configuration conf, String catalogName) {
+    if (catalogName == null) {
```

Comment:
nit, but would it be possible to invert the if-else and start with the case where `catalogName` is present? Currently it's the opposite of how the javadoc is laid out, so putting them in sync would make it a bit easier to follow the logic.

Reply:
done